src/main.rs

```diff
@@ -1,12 +1,78 @@
-use std::io::Read;
+use jotdown::{html, Parser};
+use std::{
+    error::Error,
+    ffi::OsString,
+    fs::{self, File},
+    io::{self, BufWriter},
+    process::exit,
+};
 
-fn main() {
-    let mut src = String::new();
-    std::io::stdin()
-        .read_to_string(&mut src)
-        .expect("failed to read utf-8 file");
+struct App {
+    input: Option<OsString>,
+    output: Option<OsString>,
+    format: OsString,
+}
+
+impl Default for App {
+    fn default() -> Self {
+        Self {
+            input: None,
+            output: None,
+            format: "html".into(),
+        }
+    }
+}
+
+fn parse_args() -> App {
+    let mut app = App::default();
+
+    let mut args = std::env::args_os().skip(1).peekable();
+
+    while let Some(arg) = args.next() {
+        match (arg.to_string_lossy().as_ref(), args.peek()) {
+            ("-h" | "--help", _) => {
+                eprint!(
+                    "jotdown v{}\n\n{}",
+                    env!("CARGO_PKG_VERSION"),
+                    include_str!("./help.txt")
+                );
+                exit(0);
+            }
+            ("-v" | "--version", _) => {
+                eprintln!("jotdown v{}", env!("CARGO_PKG_VERSION"));
+                exit(0);
+            }
+            ("-o" | "--output", Some(o)) => {
+                app.output = Some(o.into());
+                args.next();
+            }
+            ("-f" | "--format", Some(f)) => {
+                app.format = f.into();
+                args.next();
+            }
+            (file, _) if !file.starts_with('-') => app.input = Some(file.into()),
+            _ => {}
+        }
+    }
+
+    app
+}
+
+fn main() -> Result<(), Box<dyn Error>> {
```

Comment:
errors will cause panics now, could be nice with shorter user-friendly messages

Reply:
As of [`std::process::Termination`](https://doc.rust-lang.org/std/process/trait.Termination.html), errors won't produce a panic, but rather print the `Debug` representation of the error. You're probably right that this is a bit too rudimentary, so I'll make some friendlier messages.

Reply:
Fixed by using the `Display` impl and printing to stderr. All errors are `std::io::Error`, and I don't think there is much value in adding context.

Reply:
Yeah, this is good, much better than panic